plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -179,6 +179,10 @@ public static Table getIcebergTableWithMetadata(
     public static Map<String, Object> getIcebergTableProperties(Table icebergTable)
     {
         ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        if (icebergTable == null) {
```

Comment:
`getMaterializedViewDefinition` should make the input argument of `icebergTable` an optional and avoid calling this method when it is empty